plugins/history_plugin/history_plugin.cpp

```diff
@@ -438,6 +448,42 @@ namespace eosio {
         return result;
       }
 
+      read_write::prune_history_result read_write::prune_history( const read_write::prune_history_params& params ) const {
+         auto& chain = history->chain_plug->chain();
+         chainbase::database& db = const_cast<chainbase::database&>( chain.db() ); // Override read-only access to state DB (highly unrecommended practice!)
+
+         prune_history_result result{-1, false};
+
+         if(chain.irreversible_read_mode()) {
+            auto start_time = fc::time_point::now();
+
+            const auto& idx = db.get_index<action_history_index, by_id>();
+            auto itr = idx.lower_bound(history->last_id_pruned);
+
+            while( itr != idx.end() &&  itr->block_num <= params.height ) {
```

Comment:
This exits on the first block that is found that is larger that `params.height`. Since you are iterating `by_id` there is no order you can assume for `block_num`.